src/fetchers/git-fetcher.js

```diff
@@ -73,7 +73,17 @@ export default class GitFetcher extends BaseFetcher {
     await git.initRemote();
     await git.clone(this.dest);
 
-    let tarballInMirrorPath = this.config.getOfflineMirrorPath(this.reference);
+    // Get the tarball filename from the url
+    const {pathname} = url.parse(this.reference);
+    let tarballFilename;
+    if (pathname == null) {
+      tarballFilename = this.reference;
+    } else {
+      tarballFilename = path.basename(pathname);
+    }
+
+    let tarballInMirrorPath = this.config.getOfflineMirrorPath(tarballFilename);
```

Comment:
(sorry for a delay, I was dealing with documentation and slides debt, the fun stuff:) 
Can you come up a non breaking approach?
We would not want to force people to regenerate yarn.lock files after they started using them.